packages/edit-site/src/components/site-editor-routes/patterns.js

```diff
@@ -1,34 +1,35 @@
-/**
- * WordPress dependencies
- */
-import { privateApis as routerPrivateApis } from '@wordpress/router';
-
 /**
  * Internal dependencies
  */
 import SidebarNavigationScreenPatterns from '../sidebar-navigation-screen-patterns';
 import PagePatterns from '../page-patterns';
-import { unlock } from '../../lock-unlock';
-
-const { useLocation } = unlock( routerPrivateApis );
-
-function MobilePatternsView() {
-	const { query = {} } = useLocation();
-	const { categoryId } = query;
-
-	return !! categoryId ? (
-		<PagePatterns />
-	) : (
-		<SidebarNavigationScreenPatterns backPath="/" />
-	);
-}
+import { isClassicThemeWithStyleBookSupport } from './utils';
 
 export const patternsRoute = {
 	name: 'patterns',
 	path: '/pattern',
 	areas: {
-		sidebar: <SidebarNavigationScreenPatterns backPath="/" />,
+		sidebar( { siteData } ) {
+			const isBlockTheme = siteData.currentTheme?.is_block_theme;
+			const backPath =
+				isBlockTheme || isClassicThemeWithStyleBookSupport( siteData )
+					? '/'
+					: undefined;
+			return <SidebarNavigationScreenPatterns backPath={ backPath } />;
+		},
 		content: <PagePatterns />,
-		mobile: <MobilePatternsView />,
+		mobile( { siteData, query } ) {
+			const { categoryId } = query;
```

Comment:
Question (non-blocking): What was the reason for using `useLocation` in the previous code instead of the `query` argument? Was it just something we overlooked?

Reply:
[The previous code was written by me](https://github.com/WordPress/gutenberg/pull/69206/files#diff-9743bbd741553ca5acfe9a18a0fea3eeaaeef7168f9a3f5e51cb7722a6c9a52aR15-R24), but I think it was redundant 😅 There is no need to bother using `useLocation`.

Reply:
Sounds good 👍